compiler/ast/ast_query.nim

```diff
@@ -97,6 +97,8 @@ const
   callableDefs* = nkLambdaKinds + routineDefs
 
   nkSymChoices* = {nkClosedSymChoice, nkOpenSymChoice}
+  nkFloatKinds* = nkFloatLiterals # QUESTION remove float literals
```

Comment:
I don't think so, but maybe I'm missing some key point? Whatchya thinking?

Reply:
We have node *kinds* for everything and only float *literals*. It is a naming question, and it is a set of *node kinds*, not a set of *literals*.

Reply:
https://github.com/nim-works/nimskull/blob/837238f3cbebb6d52be6c328624f6f5598b44788/compiler/ast/ast_query.nim#L687 I think those two must be intrinsically related to each other (the collection of the enum kind sets and the `[]` named indexing operator), but for now that's just a new idea I got while elaborating on the "literals/kinds" distinction above

Reply:
For `nkFloatLiterals` vs `nkFloatKinds`, the former is literals only and the latter might include more, but I can't think of a single thing not covered by the former. The is a difference in meaning but I'm really hard pressed to think of a case where they would not be equivalent. I would check the sites and consolidate personally.

----

> https://github.com/nim-works/nimskull/blob/837238f3cbebb6d52be6c328624f6f5598b44788/compiler/ast/ast_query.nim#L687
> 
> I think those two must be intrinsically related to each other (the collection of the enum kind sets and the `[]` named indexing operator), but for now that's just a new idea I got while elaborating on the "literals/kinds" distinction above

That's exactly the types of insights that form by looking at this stuff.

I agree with the sentiment, the naming and categorization needs to make sense based on purpose and not solely because a new programmer's first instincts are to lump them together.

So to recap:
- naming needs to be more consistent
- naming needs to better convey intention
- sets need to serve a purpose(s), eg: many sites in the compiler need to handle all literals one way (`nkLiterals`) or lots of literal handling is weird about floats vs other numbers (`nkFloatLiterals`)

Doesn't have to be those precise names but I think we're on the same page. For actually changing sets carefully looking through their usage is required because there are unfortunate subtleties.